lib/ModelMixins/TableMixin.ts

```diff
@@ -496,6 +538,69 @@ function TableMixin<T extends Constructor<Model<TableTraits>>>(Base: T) {
           .colorMap;
         const valuesAsRegions = regionColumn.valuesAsRegions;
 
+        let currentTimeRows: number[];
+
+        // TODO: this is already implemented in RegionProvider.prototype.mapRegionsToIndicesInto, but regionTypes require "loading" for this to work. I think the whole RegionProvider thing needs to be re-done in TypeScript at some point and then we can move stuff into that.
```

Comment:
This sums up how I feel about this PR. 
At some point we need to sort out all of the `RegionProvider...` JavaScript stuff - which is mostly unused at the moment.
It would be nice if we could separate region-mapping complexity from `TableMixin`

Reply:
See #4715 for more info about this